packages/block-editor/src/components/block-list/use-block-props/index.js

```diff
@@ -132,7 +137,9 @@ export function useBlockProps( props = {}, { __unstableIsHtml } = {} ) {
 		} ),
 	] );
 
-	if ( blockAPIVersion < 2 ) {
+	const blockEditContext = useBlockEditContext();
+	// Ensures it warns only inside the `edit` implemenation for the block.
```

Comment:
```suggestion
	// Ensures it warns only inside the `edit` implementation for the block.
```

Reply:
Now I wonder if that was me of GitHub Copilot 😄 I'm sure it hinted "for the block" part.